### PR TITLE
cleaned up some of the code fr cloning

### PR DESCRIPTION
**Pull request by LionelEisenberg**

### CODE REVIEW

File: backend/flask_app.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def upload(files, bucket_name, version):
     for i in range(num_commits):
         #checks out commits, but ignores first time around (to get current commit)
         if i > 0:
-            proc = subprocess.Popen(("git -C " + repo_name + " checkout HEAD~1").split())
+            proc = subprocess.Popen(['git','-C', repo_name,'checkout','HEAD~1'])
```
Comment (SanatDeshpande) [inline backend/flask_app.py:64]: Why not use the git module for this one too?

File: backend/flask_app.py [hunk 2]
```diff
@@ -28,8 +28,9 @@ def last_commits():
     num_commits = int(request.args.get('num_commits', None))
     bucket_name = request.args.get('bucket_name', None)
 
-    path = os.path.dirname(os.path.realpath(__file__))
+    path = os.getcwd()
     repo_name = repo_link.split("/")[-1].split('.')[:-1][0] #this is convoluted
+    path_to_repo = path + "/" + repo_name
```
Comment (SanatDeshpande) [inline backend/flask_app.py:33]: I think `os.path.join(path, repo_name)` is the "right" way to do this

File: backend/flask_app.py (hunk 1 above)
Comment (LionelEisenberg) [reply backend/flask_app.py:64]: I looked online and the checkout for the git module I am using does not use git hooks but https hooks which makes things way slower

**Review by SanatDeshpande — APPROVED** [review]
Small stuff but I think its fine

**Pull request merged**